pkg/worker/handler/stack/detail.go

```diff
@@ -2,97 +2,68 @@ package stack
 
 import (
 	"context"
-	"regexp"
 
-	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi"
-	tagtypes "github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi/types"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
 	"github.com/xh3b4sd/tracer"
 )
 
-var (
-	// exp defines a regular expression for the ARN suffixes of nested
-	// CloudFormation stacks. Note that there is no precise definition for the
-	// amount of characters that a random hack suffix for the stack names ought to
-	// look like. So we are testing against at least 10 characters, but also
-	// accept e.g. 13 or 14.
-	//
-	//     arn:aws:cloudformation:us-west-2:995626699990:stack/server-test-FargateStack-QGXQ9XZ4J44K/165deb30-30d0-11f0-9eeb-023c6b26cb57
-	//
-	exp = regexp.MustCompile(`-[A-Z0-9]{10,}/[a-f0-9-]{36}$`)
-)
-
 type detail struct {
-	// arn is the well defined Amazon Resource Name of the CloudFormation stack.
-	arn string
+	// nam is the name of the CloudFormation stack.
+	nam string
+	// sta is the raw CloudFormation stack status, e.g. UPDATE_COMPLETE.
+	sta types.StackStatus
 }
 
-// detail finds all CloudFormation stack ARNs that are tagged with the
+// detail finds all CloudFormation stack statuses that are tagged with the
 // "environment" that matches Specta's runtime configuration. In other words, if
 // Specta is running in "staging", then detail() will find all CloudFormation
 // stacks labelled with the resource tags environment=staging.
 func (h *Handler) detail() ([]detail, error) {
 	var err error
 
-	var inp *resourcegroupstaggingapi.GetResourcesInput
+	// Create a new paginator for each instrumentation cycle in order to guarantee
+	// data integrity per loop.
+	var pag *cloudformation.DescribeStacksPaginator
 	{
-		inp = &resourcegroupstaggingapi.GetResourcesInput{
-			ResourceTypeFilters: []string{"cloudformation:stack"},
-			TagFilters: []tagtypes.TagFilter{
-				{
-					Key:    aws.String("environment"),
-					Values: []string{h.env.Environment},
-				},
-			},
-		}
-	}
-
-	var out *resourcegroupstaggingapi.GetResourcesOutput
-	{
-		out, err = h.tag.GetResources(context.Background(), inp)
-		if err != nil {
-			return nil, tracer.Mask(err)
-		}
+		pag = cloudformation.NewDescribeStacksPaginator(h.cfc, &cloudformation.DescribeStacksInput{})
```

Comment:
Instead of using the tagging API, we use the appropriate paginator in order to circumvent any rate-limiting errors.